src/geo_key_directory.rs

```diff
@@ -0,0 +1,647 @@
+                        &double_params_data,
+                        key_tag,
+                        location_tag,
+                        *count,
+                        *value_or_offset,
+                    )?
+                    .into()
+                }
+                GeoKeyDirectoryTag::ProjScaleAtCenter => {
+                    directory.proj_scale_at_center = Self::get_double(
+                        &double_params_data,
+                        key_tag,
+                        location_tag,
+                        *count,
+                        *value_or_offset,
+                    )?
+                    .into()
+                }
+                GeoKeyDirectoryTag::ProjAzimuthAngle => {
+                    directory.proj_azimuth_angle = Self::get_double(
+                        &double_params_data,
+                        key_tag,
+                        location_tag,
+                        *count,
+                        *value_or_offset,
+                    )?
+                    .into()
+                }
+                GeoKeyDirectoryTag::ProjStraightVertPoleLong => {
+                    directory.proj_straight_vert_pole_long = Self::get_double(
+                        &double_params_data,
+                        key_tag,
+                        location_tag,
+                        *count,
+                        *value_or_offset,
+                    )?
+                    .into()
+                }
+                GeoKeyDirectoryTag::Vertical => {
+                    directory.vertical =
+                        Self::get_short(key_tag, location_tag, *count, *value_or_offset)?.into()
+                }
+                GeoKeyDirectoryTag::VerticalCitation => {
+                    directory.vertical_citation = Self::get_string(
+                        &ascii_params_data,
+                        key_tag,
+                        location_tag,
+                        *count,
+                        *value_or_offset,
+                    )?
+                    .into()
+                }
+                GeoKeyDirectoryTag::VerticalDatum => {
+                    directory.vertical_datum =
+                        Self::get_short(key_tag, location_tag, *count, *value_or_offset)?.into()
+                }
+                GeoKeyDirectoryTag::VerticalUnits => {
+                    directory.vertical_units =
+                        Self::get_short(key_tag, location_tag, *count, *value_or_offset)?.into()
+                }
+            }
+        }
+
+        Ok(directory)
+    }
+
+    fn get_short(
+        key_tag: GeoKeyDirectoryTag,
+        location_tag: Option<Tag>,
+        count: u16,
+        offset: u16,
+    ) -> TiffResult<u16> {
+        if location_tag.is_some() {
+            return Err(TiffError::FormatError(TiffFormatError::Format(format!(
+                "Key `{key_tag:?}` did not have the expected value type."
+            ))));
+        }
+
+        if count != 1 {
```

Comment:
Could use a ref here instead, and change all the `*count` to `count` above?

```suggestion
        count: &u16,
        offset: u16,
    ) -> TiffResult<u16> {
        if location_tag.is_some() {
            return Err(TiffError::FormatError(TiffFormatError::Format(format!(
                "Key `{key_tag:?}` did not have the expected value type."
            ))));
        }

        if count != &1 {
```

Reply:
I can do that, but I'm unsure whether I completely understand the rationale. I think you want me to change it to a reference because of the contract of the equality operator: `fn eq(&self, &other: Self): bool`. Is there a performance impact to this, or is it just a matter of style? I can imagine the compiler will optimize this anyway. For me, `*count == 1` is more intuitive than `count == &1`, as in my mind, I'm comparing values rather than references. Changing from `count: u16` to `count: &u16` also consumes more memory on the stack, four times as much on 64-bit systems. Anyway, I'm happy to change it if you want me to; I just wanted to share my thoughts. Is there any best-practices document you follow that I can relate to?

Reply:
I'd love to hear rationale either way or see some resources on this, but for types that implement `Copy` I think it's simpler to not use a reference in the signature parameter

Reply:
Ah ok, I was approaching this more as a matter of style rather than performance. Still learning Rust on the go, and didn't realize that references to u16 would be stored as 64-bit (!!), so happy for you to stick with `*count == 1`.

Reply:
I'm no expert on this topic either, but inspecting the code via [Compiler Explorer](https://godbolt.org) might reveal more insights.